### PR TITLE
Update esa_landcover to use Dataset class

### PR DESCRIPTION
**Pull request by jacobwhall**
- esa_landcover now has one main script with a class definition and helper functions
- INI config file with settings like directories to read and write, whether to overwrite files, which years to process
- some Python modernization, e.g. using pathlib

### CODE REVIEW

File: esa_landcover/main.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        
+        logger = self.get_logger()
+
+        if year in self.v207_years:
+            version = "v2.0.7cds"
+        elif year in self.v211_years:
+            version = "v2.1.1"
+        else:
+            version = "v2.1.1"
+            logger.warning(f"Assuming that {year} is v2.1.1")
```
Comment (jacobwhall) [inline esa_landcover/main.py:121]: This case used to raise an exception, but I've made it a warning.

**Comment by jacobwhall** [comment]
I have now tested this deployment, and it ran smoothly

File: esa_landcover/deploy.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+# Driver Code
+flow = flow_import(module_name, flow_name)
+
+
+# create and load storage block
+
+block_name = "geo-datasets-github"
+block_repo = "https://github.com/aiddata/geo-datasets.git"
+block_reference = 'develop' # branch or tag
```
Comment (jacobwhall) [inline esa_landcover/deploy.py:26]: Please note that we'll need to update this reference if/when we merge this branch into master

**Comment by sgoodm** [comment]
Looks good. Moving this into a dedicated esa_landcover branch for integration testing. Main item to address is adding deployment / run config args

**Pull request merged**